src/Sparrow/Compression/LZ4.cs

```diff
@@ -79,6 +79,35 @@ protected struct LZ4_stream_t_internal
             public uint initCheck;
         }
 
+        public long Encode64LongBuffer(
+            byte* input,
+            byte* output,
+            long inputLength,
+            long outputLength,
+            int acceleration = ACCELERATION_DEFAULT)
+        {
+            if (inputLength < int.MaxValue && outputLength < int.MaxValue)
+            {
+                return Encode64(input, output, (int)inputLength, (int)outputLength, acceleration);
+            }
+
+            long totalOutputSize = 0;
+            long pos = 0;
+            while (pos < inputLength)
+            {
+                int partInputLength = int.MaxValue;
+                if (pos + partInputLength > inputLength)
+                    partInputLength = (int)(Int32.MaxValue - pos);
+
+                int partOutputLength = int.MaxValue;
+                totalOutputSize += Encode64(input, output, partInputLength, partOutputLength, acceleration);
```

Comment:
In here, you are always compressing the first part of the data, and never the latter part.
The `input` and `output` values aren't change, and that is what it is using.